app/src/main/java/org/fossasia/susi/ai/data/SkillListingModel.kt

```diff
@@ -73,9 +73,12 @@ class SkillListingModel : ISkillListingModel {
 
     override fun cancelFetch() {
         try {
-            authResponseCallGroups.cancel()
-            authResponseCallSkills.cancel()
-            authResponseCallMetrics.cancel()
+            while (!authResponseCallGroups.isCanceled)
+                authResponseCallGroups.cancel()
+            while (!authResponseCallSkills.isCanceled)
+                authResponseCallSkills.cancel()
+            while (!authResponseCallMetrics.isCanceled)
+                authResponseCallMetrics.cancel()
```

Comment:
This is strangest use of while loop I have ever seen!

Reply:
Can you please suggest what can be done instead

Reply:
There is no need of while loop. Simply  solve the memory leak using common practices

Reply:
Can I use Handler().postDelayed

Reply:
No

Reply:
Can I use boolean to check if the fetch is called or not

Reply:
You simply need to return from the callbacks if the presenter is destroyed